simtools/simtel/simtel_events.py

```diff
@@ -1,194 +1,261 @@
-import copy
+import math
 import logging
 import numpy as np
-import matplotlib.pyplot as plt
-from matplotlib.backends.backend_pdf import PdfPages
+from copy import copy
 
-from eventio import EventIOFile, Histograms
-from eventio.search_utils import yield_toplevel_of_type
+import astropy.units as u
 
-__all__ = ['SimtelHistograms']
+from eventio.simtel import SimTelFile
 
 
-class BadHistogramFormat(Exception):
+__all__ = ['SimtelEvents']
+
+
+class InconsistentInputFile(Exception):
     pass
 
 
-class SimtelHistograms:
+class SimtelEvents:
     '''
-    This class handle sim_telarray histograms.
-    Histogram files are handled by using eventio library.
+    This class handle sim_telarray events.
+    sim_telarray files are read with eventio package,
 
     Methods
     -------
     plotAndSaveFigures(figName)
         Plot all histograms and save a single pdf file.
+
+
+    Attributes
+    ----------
+    inputFiles: list
+        List of sim_telarray files.
+    summaryEvents: dict
+        Arrays of energy and core radius of events.
     '''
 
-    def __init__(
-        self,
-        histogramFiles
-    ):
+    def __init__(self, inputFiles=None):
         '''
-        SimtelHistograms
+        SimtelEvents
 
         Parameters
         ----------
-        histogramFiles: list
-            List of sim_telarray histogram files (str of Path).
-
+        inputFiles: list
+            List of sim_telarray output files (str of Path).
         '''
         self._logger = logging.getLogger(__name__)
-        self._histogramFiles = histogramFiles
+        self.loadInputFiles(inputFiles)
+        self.loadHeaderAndSummary()
```

Comment:
This should be called only if `inputFiles` is not `None` or its length is longer than 1.

Reply:
I made that change now, but check if you prefer to check length instead.

Reply:
Perfect.

I actuallly did something a bit fancier. I added a property numberOfFiles and I check if numberOfFiles > 0.